### PR TITLE
Add some granularity to syntax highlighting.

### PR DESCRIPTION
**Pull request by omerbenamram**
Hi,

I wanted to start using `rust-analyzer` a bit more frequently - one of the main blockers for me so far was the highlighting.

I just discovered it's possible to override the default colors with `ralsp.<something>` setting without waiting for #2061!

However, the current implementation was lumping a bunch of different tokens into `type` and `literal`.
The golden standard IMO is what Clion is currently doing (and is my current daily driver for rust).

Clion allows users to control the coloring for specific literal kinds, and the default is to distinguish between them (numerics get a different color from strings, and special colors for bytestrings).

I've also splitted the builtin types, which are also allowed to be highlighted speratly.
My goal is to match the default experience I'm getting with clion.
The only blockers now I think is that `rust-analyzer` doesn't corrently infer types in some situations, so the highlighting information is incorrect in those cases.

This is what it looks like so far (with colors overriden to match clion's theme):
image

If there are any other changes you feel is necessary let me know.

I did leave the default colors to match the current behavior, since I'm not familiar with the colors for this theme, I added some random (different) colors in the test to check that it indeed was working.

### CODE REVIEW

**Comment by matklad** [comment]
LGTM, but CI is failing?

**Comment by omerbenamram** [comment]
@matklad Commited the changes to the as the rainbow highlighting test aswell, should pass now.

**Comment by matklad** [comment]
bors merge

File: editors/code/package.json [hunk 1]
```diff
@@ -457,7 +484,43 @@
...
+                "description": "Color for built-in types (&str, bool, u16, u32)",
+                "defaults": {
+                    "dark": "#4EC9B0",
+                    "light": "#267F99",
+                    "highContrast": "#4EC9B0"
+                }
+            },
+            {
+                "id": "ralsp.type.lifetime",
+                "description": "Color for `Self` param type",
```
Comment (lnicola) [inline editors/code/package.json]: Typo

**Comment by ice1000** [comment]
@omerbenamram How can I set my highlighting to the one like yours (CLion style)?

**Comment by omerbenamram** [comment]
@ice1000 this is what I have for settings.
```
    "workbench.colorCustomizations": {
        "ralsp.keyword": "#CC7832",
        "ralsp.builtin": "#CC7832",
        "ralsp.keyword.control": "#CC7832",
        "ralsp.macro": "#4EADE5",
        "ralsp.attribute": "#BBB529",
        "ralsp.variable": "#A9B7C6",
        "ralsp.variable.mut": "#A9B7C6",
        "ralsp.parameter": "#A9B7C6",
        "ralsp.string": "#6A8759",
        "ralsp.function": "#FFC66D",
        "ralsp.module": "#A9B7C6",
        "ralsp.literal.numeric": "#6898BB",
        "ralsp.literal.char": "#6A8759",
        "ralsp.constant": "#9876AA",
        "ralsp.type": "#A9B7C6",
        "ralsp.type.param": "#20999D",
        "ralsp.type.lifetime": "#20999D",
        "ralsp.type.self": "#CC7832",
        "ralsp.type.builtin": "#CC7832",
        "ralsp.field": "#9876AA"
    },
```

**Comment by ice1000** [comment]
@omerbenamram Thank you!!

**Pull request merged**